thetis/diagnostics.py

```diff
@@ -2,24 +2,30 @@
 Classes for computing diagnostics.
 """
 from .utility import *
+from .configuration import *
 
 
-__all__ = ["VorticityCalculator2D", "HessianRecoverer2D"]
+__all__ = ["VorticityCalculator2D", "HessianRecoverer2D", "DynamicPressureCalculator"]
 
 
-class VorticityCalculator2D(object):
+class VorticityCalculator2D(FrozenHasTraits):
     r"""
     Linear solver for recovering fluid vorticity.
 
     It is recommended that the vorticity is sought
     in :math:`\mathbb P1` space.
-
-    :arg uv_2d: horizontal velocity :class:`Function`.
-    :arg vorticity_2d: :class:`Function` to hold calculated vorticity.
-    :kwargs: to be passed to the :class:`LinearVariationalSolver`.
     """
+    uv_2d = FiredrakeVectorExpression(
+        Constant(as_vector([0.0, 0.0])), help='Horizontal velocity').tag(config=True)
```

Comment:
Using traits here is nice as we can check the inputs dynamically. Having to define a default value is however a bit odd though but I guess we cannot avoid that.